From 2ee809d53de23a04a48e3fbfb48eaf7a55992a77 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Mon, 5 Jul 2021 12:31:40 +0200
Subject: [PATCH] boot src cmd: custom scan_overlays cmd

- Modify "scan_overlays" of boot.scr.cmd file to add extra space with
fdt resize operation. Without this additional space, device tree
overlays application can fails.
- Add possibility to specify the DTO to apply using "overlay" u-boot env
var. If the variable is defined, content of overlays.txt is ignored.
---
 recipes-bsp/u-boot/u-boot-stm32mp-extlinux/boot.scr.cmd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boot.scr.cmd b/boot.scr.cmd
index eeedb2e..731ad2d 100644
--- a/boot.scr.cmd
+++ b/boot.scr.cmd
@@ -23,7 +23,7 @@ env set boot_m4fw 'rproc init; rproc load 0 ${m4fw_addr} ${filesize}; rproc star
 env set scan_m4fw 'if test -e ${devtype} ${devnum}:${distro_bootpart} ${m4fw_name};then echo Found M4 FW $m4fw_name; if load ${devtype} ${devnum}:${distro_bootpart} ${m4fw_addr} ${m4fw_name}; then run boot_m4fw; fi; fi;'
 
 # management of overlay
-env set scan_overlays 'if test -e ${devtype} ${devnum}:${distro_bootpart} /overlays/overlays.txt; then env set fdt_addr ${fdt_addr_r}; if load ${devtype} ${devnum}:${distro_bootpart} ${fdt_addr} ${fdtfile}; then fdt addr ${fdt_addr}; fdt resize; setexpr fdtovaddr ${fdt_addr} + C0000; if load ${devtype} ${devnum}:${distro_bootpart} ${fdtovaddr} /overlays/overlays.txt && env import -t ${fdtovaddr} ${filesize} && test -n ${overlay}; then echo loaded overlay.txt: ${overlay}; for ov in ${overlay}; do echo overlaying ${ov}...; load ${devtype} ${devnum}:${distro_bootpart} ${fdtovaddr} /overlays/${ov}.dtbo && fdt apply ${fdtovaddr}; done; fi; fi; fi;'
+env set scan_overlays 'if test -e ${devtype} ${devnum}:${distro_bootpart} /overlays/overlays.txt || test -n ${overlay}; then env set fdt_addr ${fdt_addr_r}; if load ${devtype} ${devnum}:${distro_bootpart} ${fdt_addr} ${fdtfile}; then fdt addr ${fdt_addr}; fdt resize; setexpr fdtovaddr ${fdt_addr} + C0000; if test -z "${overlay}"; then load ${devtype} ${devnum}:${distro_bootpart} ${fdtovaddr} /overlays/overlays.txt; env import -t ${fdtovaddr} ${filesize}; fi; if test -n ${overlay}; then echo loaded overlay.txt: ${overlay}; for ov in ${overlay}; do echo overlaying ${ov}...; load ${devtype} ${devnum}:${distro_bootpart} ${fdtovaddr} /overlays/${ov}.dtbo && fdt resize ${filesize} && fdt apply ${fdtovaddr}; done; fi; fi; fi;'
 
 # Update the DISTRO command to search in sub-directory and load M4 firmware
 env set boot_prefixes "/${boot_device}${boot_instance}_"
-- 
2.17.1

